internal/iapl/rbac.go

```diff
@@ -0,0 +1,222 @@
+package iapl
+
+import (
+	"go.infratographer.com/permissions-api/internal/types"
+)
+
+const (
+	// RoleOwnerRelation is the name of the relationship that connects a role to its owner.
+	RoleOwnerRelation = "owner"
+	// RoleOwnerMemberRoleRelation is the name of the relationship that connects a resource
+	// to a role that it owns
+	RoleOwnerMemberRoleRelation = "member_role"
+	// AvailableRoleRelation is the name of the relationship that list all roles
+	// that are available for a given resource
+	AvailableRoleRelation = "avail_role"
```

Comment:
This seems like it could be named a little better, since it's actually used as an action everywhere, not as a relation. Maybe this would be better? Not sure.

```suggestion
	// UseRoleAction is the name of the action that allows the use of a role
	// for listing, binding, etc in the context of a given resource
	UseRoleAction = "iam_use_role"
```